lib/index.js

```diff
@@ -0,0 +1,148 @@
+const logger = require('./logger');
+
+require('./config.js');
+
+const {PaymentStatus, Errors} = paymentService;
+
+const {ensureAuthenticated} = brPassport;
+const {config} = bedrock;
+const {BedrockError} = bedrock.util;
+
+bedrock.events.on('bedrock-express.configure.routes', app => {
+  const cfg = config['payment-http'];
+  const {basePath: PAYMENT_ENDPOINT} = cfg.routes;
+  // GET /payment/credentials
+  // GET the api keys/ auth tokens necessary to implement charges client side.
+  app.get(
+    `${PAYMENT_ENDPOINT}/credentials`,
+    ensureAuthenticated,
+    asyncHandler(async (req, res) => {
+      const {service} = req.query;
+      if(!service) {
+        throw new BedrockError(
+          'You must provide a service in the query', Errors.Data);
+      }
+      logger.info(`Getting ${service} credentials`);
+      const data = await paymentService.getGatewayCredentials({service});
+      if(!data) {
+        throw new BedrockError(
+          `No Credentials Found for ${service}`, Errors.Data);
+      }
+      return res.status(200).json(data);
+    })
+  );
+
+  // GET /payment
+  // GET all payments for an account.
+  app.get(
+    PAYMENT_ENDPOINT,
+    ensureAuthenticated,
+    asyncHandler(async (req, res) => {
+      // this is so a user can not request
+      // another user's payments.
+      const {id: creator} = req.user.account;
+      // check for existing Payments.
+      const query = {creator};
+      const payments = await paymentService.db.findAll({query});
+      return res.status(200).json(payments);
+    })
+  );
+
+  // POST /payment
+  // Creates a new Payment.
+  // The client will need to post the payment they intend to use.
+  app.post(
+    PAYMENT_ENDPOINT,
+    ensureAuthenticated,
+    validate({body: 'payment.creating'}),
+    asyncHandler(async (req, res) => {
+      const {id: creator} = req.user.account;
+      const {payment: paymentData} = req.body;
+      logger.debug('CREATE PAYMENT CALLED', {paymentData});
+      const processingPayments = await paymentService.db.findAll(
+        {query: {creator, status: PaymentStatus.PROCESSING}});
+      // this can cause issues if a payment fails while processing.
+      if(processingPayments.length > 0) {
+        throw new BedrockError(
+          'Can not create a new payment if you have processing payments.',
+          Error.Constraint, {httpStatusCode: 409, public: true});
+      }
+      // check for existing pending Payments.
+      const pendingPayments = await paymentService.db.findAll(
+        {query: {creator, status: PaymentStatus.PENDING}});
+      if(pendingPayments.length > 0) {
+        // this is where most of the concurrency issues appear.
+        const {updatedOrder, payment} = await paymentService.
+          updatePendingPayment({pendingPayments, paymentData});
+        return res.status(200).json({order: updatedOrder, payment});
+      }
+      logger.debug('CREATING NEW PAYMENT', {creator, paymentData});
```

Comment:
If this logging should remain, make it into a properly cased sentence with punctuation. Consider any security implications of this data being in the log, anything sensitive in there?
```suggestion
      logger.debug('Creating a new payment.', {creator, paymentData});
```

Reply:
Addressed here: https://github.com/digitalbazaar/bedrock-payment-http/pull/1/commits/3dbb7eb8ce3be7e2d9e6d23ef1efbb2c6a1b31a1